server/neptune/workflows/internal/deploy/revision/queue/queue.go

```diff
@@ -54,6 +54,10 @@ func (q *Deploy) Pop() (terraform.DeploymentInfo, error) {
 	return q.queue.Pop()
 }
 
+func (q *Deploy) Scan(priority PriorityType) []terraform.DeploymentInfo {
```

Comment:
I initially did not want to expose priority type,  My recommendation here was to have Scan() return a combined queue with manual deployments at the front and merged at the back.  Is there a reason why you didn't do it this way?

Reply:
We don't need to scan to whole queue, just the high priority elements. I can change it though.